Not able to compile for QNX

## Required information

**Operating system:**
Ubuntu 18.04 LTS

**Compiler version:**
QCC 5.4.0

I am trying to build Iceoryx for QNX by sourcing QNX SDP 7.0 in my terminal, through the provided script and the toolchain file for x86_64. But I am getting the following error
`/home/indramani/iceoryx/iceoryx_hoofs/platform/qnx/source/grp.cpp:21:53: error: 'getgrouplist' was not declared in this scope`

--- a/iceoryx_hoofs/platform/qnx/include/iceoryx_hoofs/platform/grp.hpp
+++ b/iceoryx_hoofs/platform/qnx/include/iceoryx_hoofs/platform/grp.hpp
@@ -18,6 +18,7 @@
 #define IOX_HOOFS_QNX_PLATFORM_GRP_HPP
 
 #include <grp.h>
+#include <unistd.h>
 
 int iox_getgrouplist(const char* user, gid_t group, gid_t* groups, int* ngroups);
 

--- a/iceoryx_posh/include/iceoryx_posh/popo/trigger_handle.hpp
+++ b/iceoryx_posh/include/iceoryx_posh/popo/trigger_handle.hpp
@@ -37,7 +37,8 @@ namespace popo
 class TriggerHandle
 {
   public:
-    TriggerHandle() noexcept = default;
+    /// @note explicitly implemented for MSVC and QNX
+    TriggerHandle() noexcept;
 
     /// @brief Creates a TriggerHandle
     /// @param[in] conditionVariableDataRef reference to a condition variable data struct

--- a/iceoryx_posh/source/popo/trigger_handle.cpp
+++ b/iceoryx_posh/source/popo/trigger_handle.cpp
@@ -21,6 +21,11 @@ namespace iox
 {
 namespace popo
 {
+/// explicitly implemented for MSVC and QNX
+TriggerHandle::TriggerHandle() noexcept
+{
+}
+
 TriggerHandle::TriggerHandle(ConditionVariableData& conditionVariableData,
                              const cxx::MethodCallback<void, uint64_t> resetCallback,
                              const uint64_t uniqueTriggerId) noexcept

